### ISSUE TITLE
Incorrect pipeline total element count for `bst track`

### ISSUE THREAD

**Issue by cs-shadow** [cs-shadow/buildstream#52]
In GitLab by @tristanvb on Jul 19, 2017, 07:32

When running `bst track` without `--deps all`, the pipeline status area still shows a total element count inclusive of the track target's elements, even though the intention (and behavior) is only to track a single element.

This may also be true in other scenarios like with `bst fetch`, the counter should be changed to be based on the real list of elements which are queued for scheduling.

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Jul 24, 2017, 10:28

As far as I can tell, it currently shows both, the number of session elements (i.e., 1 in the above `bst track` case) and the total number of elements (always counting all dependencies). It may indeed be confusing to show the total element count in the status area, especially as it's formatted like progress (1/42).

I propose to drop the total element count from the status area and add a counter of processed elements instead, i.e., how many elements have passed through the whole pipeline. And keep the number of session elements. Or did you have something else in mind?

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on Jul 26, 2017, 14:48

Ok so this is a bit confusing.

What you say is literally correct (about pipeline total vs session count).

However, the use case is basically something we derived from YBD's counter, we want to know:
* How many total elements there are for a build
* How many of those elements needed to be built in this session
* How many elements have we built so far in this session

What I have in mind for the status area is to reflect that information, the last bit is represented in the scheduler queues, and the first two are in the first status line.

In this light, I suppose that it does not make sense to include the "total elements" in the status area for any bst commands where this does not make sense:
* It makes sense for a build
* It does not make sense at all for `bst track`, because `bst track` will do unconditional processing on each element
* It *might* make sense for `bst fetch`, if we first filter out any already consistent sources from the session elements fed to the scheduler (i.e. fetching 8/26 elements makes sense)
* It also *might* make sense for a potential `bst push` command, where we can first consult the summary and say "Pushing 3/26 elements", by filtering out any elements which already exist on the server from the scheduler input.
